### DISCUSSION TITLE
serve plain text files on github pages

### DISCUSSION THREAD

**Discussion by shug0131** [community/community#145557] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I'm trying to serve plain text files on github pages.  The filenames  finish with a non-standard file extension.  *.Rout .  So clicking the link in the browser asks to save the files, rather than view them.   If I name with *.txt then I can view as desired.

Is there a way to remove the reliance on *.txt?  Or some documentation to find out what the constraints actually are.

**Comment by ELHart05** [comment, marked as answer]
`.txt` will definetly work because `.txt` extension is recognized as a plain text file where browsers automatically display it. also Github Pages serves files based on their [MIME type](https://developer.mozilla.org/en-US/docs/Web/HTTP/MIME_types/Common_types), and .Rout files aren't associated with a the MIME type that browsers recognize for viewing text content.

The workaround that I suggest is to create an HTML page where you basically have execute a fetch function that GETs the file with the extension then print it as text on the body like this

 ```html
 <body>
    <div id="file-content">Loading file...</div>

    <script>
        fetch('./X.Route')
            .then(response => {
                console.log(response)
                return response.text() //this gonna extract the content of the file
            })
            .then(data => {
                document.getElementById('file-content').textContent = data;
            })
            .catch(error => {
                document.getElementById('file-content').textContent = `Error loading file: ${error}`;
            });
    </script>
</body>
 ```
 
 Note that you can add a feature like having a query that gets the file name and then do the fetch based on it!
